client/src/main/java/com/microsoft/durabletask/DurableTaskGrpcClient.java

```diff
@@ -173,7 +183,16 @@ public OrchestrationMetadata waitForInstanceCompletion(String instanceId, Durati
         TaskHubSidecarServiceBlockingStub grpcClient = this.sidecarClient.withDeadlineAfter(
                 timeout.toMillis(),
                 TimeUnit.MILLISECONDS);
-        GetInstanceResponse response = grpcClient.waitForInstanceCompletion(request);
+
+        GetInstanceResponse response;
+        try {
+            response = grpcClient.waitForInstanceCompletion(request);
+        } catch (StatusRuntimeException e) {
+            if (e.getStatus().getCode() == Status.Code.DEADLINE_EXCEEDED) {
+                throw new TimeoutException("Orchestration instance completion timeout reached");
```

Comment:
Same comment as before:
```suggestion
                throw new TimeoutException("Orchestration instance completion timeout reached.");
```